src/stable_hasher.rs

```diff
@@ -85,23 +148,23 @@ impl StableHasher {
     /// To be used in-place of [`Hasher::finish`].
     #[inline]
     #[must_use]
-    pub fn finish<W: StableHasherResult>(self) -> W {
-        W::finish(self.state.finish128())
+    pub fn finish<W: StableHasherResult<Hash = H::Hash>>(self) -> W {
```

Comment:
I'm wondering if we need `StableHasherResult` at all. Would we lose anything if we just made this:
```rust
pub fn finish<R: From<H::Hash>>(self) -> R {
    R::from(self.state.finish())
}
```

Reply:
Depends on what we want; if a hasher returned `u64` would we want the return from `finish` to be `u128`[^1]? IMO I don't think we want to allow those kind of conversions to happen.

I think it's better to explicitly implement the trait, in particular since it might conflict with regular usage of the inner hash type. In terms of documentation it might also be better to only have explicitly implemented trait as it would be easier to see which one might be used.

[^1]: because of `impl From<u64> for u128`

Reply:
Yeah, that makes sense. Let's keep the explicit trait.